utility_analysis_new/combiners.py

```diff
@@ -41,20 +43,107 @@ class CountUtilityAnalysisMetrics:
     noise_kind: pipeline_dp.NoiseKind
 
 
+@dataclass
+class SumOfRandomVariablesMoments:
+    """Stores moments of sum of random independent random variables."""
+    count: int
+    expectation: float
+    variance: float
+    third_central_moment: float
+
+    def __add__(
+            self, other: 'SumOfRandomVariablesMoments'
+    ) -> 'SumOfRandomVariablesMoments':
+        return SumOfRandomVariablesMoments(
+            self.count + other.count, self.expectation + other.expectation,
+            self.variance + other.variance,
+            self.third_central_moment + other.third_central_moment)
+
+
+def _probabilities_to_moments(
+        probabilities: List[float]) -> SumOfRandomVariablesMoments:
+    """Computes moments of sum of independent bernoulli random variables."""
+    exp = np.sum(probabilities)
+    var = np.sum([p * (1 - p) for p in probabilities])
+    third_central_moment = np.sum(
+        [p * (1 - p) * (1 - 2 * p) for p in probabilities])
+
+    return SumOfRandomVariablesMoments(len(probabilities), exp, var,
+                                       third_central_moment)
+
+
+@dataclass
+class PartitionSelectionAccumulator:
+    """Stores data needed for computing probability of keeping the partition.
+
+    Args:
+        probabilities: probabilities that each specific user contributes to the
+        partition after contribution bounding.
+        moments: contains moments of the sum of independent random
+        variables, which represent whether user contributes to the partition.
+
+    Those variables are set mutually exclusive. If len(probabilities) <= 
+    MAX_PROBABILITIES_IN_ACCUMULATOR then 'probabilities' are used otherwise
```

Comment:
Just a thought - MAX_PROBABILITIES_IN_ACCUMULATOR is kind of like an enable flag in a sense. Do you think the client may want to have more control to tune this or turn on/off? They could override it?

Reply:
This const looks too implementation specific, it's unlikely useful for clients. If some clients will want this, we can expose as a flag, otherwise let's keep API small :)